napari/_qt/tests/test_qt_range_slider_popup.py

```diff
@@ -0,0 +1,40 @@
+import numpy as np
+
+from napari._qt.qt_range_slider_popup import QRangeSliderPopup
+
+
+def test_range_slider_popup(qtbot):
+    initial = np.array([100, 400])
+    range_ = np.array([0, 500])
+    precision = 2
+    pop = QRangeSliderPopup(
+        horizontal=True,
+        precision=precision,
+        initial_values=initial,
+        data_range=range_,
+        step_size=1,
+    )
+    pop.show()
+
+    # make sure labels are correct
+    assert float(pop.curmin_label.text()) == initial[0]
+    assert float(pop.curmax_label.text()) == initial[1]
+    assert np.all(pop.slider.range == range_)
+
+    # make sure setting the slider updates the labels
+    pop.slider.setValues((10, 20))
+    assert float(pop.curmin_label.text()) == 10
+    assert float(pop.curmax_label.text()) == 20
+    # and vice versa
+    pop.curmin_label.setText('100')
+    pop.curmax_label.setText('300')
+    pop.curmin_label.editingFinished.emit()
+    pop.curmax_label.editingFinished.emit()
+    assert np.all(pop.slider.values() == (100, 300))
+
+    # make sure changing the range labels updates the slider
```

Comment:
As above: it's nicer if these tests can be broken up into separate test functions (then these comments become docstrings). You can use a fixture to create a repeated starting condition...